twine/repository.py

```diff
@@ -108,7 +124,7 @@ def set_client_certificate(self, clientcert: Optional[str]) -> None:
             self.session.cert = clientcert
 
     def register(self, package: package_file.PackageFile) -> requests.Response:
-        data = package.metadata_dictionary()
+        data = dict(**package.metadata_dictionary())
```

Comment:
I reverted this change and the tests still pass. Why `dict(**)`? It would be better if the result from `package.metadata_dictionary()` did not require the caller to wrap the result in `dict(**)` or `dict()`. How can that be achieved?

Reply:
The `**` are indeed not necessary. They are a remnant of a previous approach. The `dict()` is just to make mypy happy: the code adds keys to the dictionary outside the typed keys and mypy does not like that. I can replace `dict()` with a cast if you prefer.

Reply:
I restructured the code to do not require the type casting anymore.